src/inspect_ai/tool/_tool_support_helpers.py

```diff
@@ -0,0 +1,173 @@
+    """See: https://www.jsonrpc.org/specification#error_object"""
+
+    code: int
+    message: str
+    data: object | None = None
+
+
+class JSONRPCErrorResponse(JSONRPCResponseBase):
+    error: JSONRPCError
+
+
+class JSONRPCResponse(RootModel[JSONRPCSuccessResponse | JSONRPCErrorResponse]):
+    pass
+
+
+BaseModelT = TypeVar("BaseModelT", bound=BaseModel)
+
+
+id_generator = count(666)
+
+
+async def exec_sandbox_rpc(
+    sandbox: SandboxEnvironment,
+    method: str,
+    params: dict[str, object] | tuple[object, ...],
+    result_cls: Type[BaseModelT],
+    timeout: int | None = None,
+    user: str | None = None,
+) -> BaseModelT:
+    """
+    Execute a JSON-RPC command to a sandbox environment.
+
+    Note that the JSON RPC request is sent to the exec'ed program via stdin.
+
+    Args:
+      sandbox (SandboxEnvironment): The sandbox environment to execute the command in.
+      method (str): The JSON-RPC method to call.
+      params (dict[str, object] | tuple[object, ...]): The parameters for the JSON-RPC method.
+      result_cls (Type[BaseModelT]): The class to use for parsing the result.
+      timeout (int | None, optional): The timeout for the execution. Defaults to None.
+      user: Optional username or UID to run the command as.
+
+    Returns:
+      BaseModelT: The parsed result of the JSON-RPC call.
+
+    Raises:
+      RuntimeError: If the sandbox execution fails or if there is an error in the JSON-RPC response.
+      ToolParsingError: If the JSON-RPC response contains a specific error code indicating a parsing error.
+    """
+    exec_result = await sandbox.exec(
+        [SANDBOX_CLI, "exec"],
+        input=_create_json_rpc_request(method, params),
+        timeout=timeout,
+        user=user,
+    )
+
+    if not exec_result.success:
+        raise RuntimeError(
+            f"Sandbox.exec failure executing {_rpc_call_description(method, params)}: {exec_result.stderr}"
+        )
+
+    match _parse_json_rpc_response(exec_result.stdout, result_cls):
+        case JSONRPCError(-32601, message=message) | JSONRPCError(
+            -32602, message=message
+        ):
+            raise ToolParsingError(message)
+        case JSONRPCError(-32000, message=message):
+            raise ToolError(message)
+        case JSONRPCError(code=code, message=message):
+            raise RuntimeError(
+                f"Error executing tool command {_rpc_call_description(method, params)}: {code=} {message}"
+            )
+        # case result_cls() as model: yields a mypy error since it has narrowed model down
+        # to BaseModel and not BaseModelT. ???
+        case model if isinstance(model, result_cls):
+            return model
+        case not_possible:
+            raise RuntimeError(
+                f"Error executing tool command {_rpc_call_description(method, params)}: {not_possible}"
+            )
+
+
+SANDBOX_CLI = "inspect-tool-support"
+INSPECT_TOOL_SUPPORT_IMAGE_DOCKERHUB = "aisiuk/inspect-tool-support"
```

Comment:
@jjallaire, we didn't discuss the name of the image(s) we'll publish. TBH, I think we should publish one for each of - ubuntu, debian, kali. I have separate, slightly different, Dockerfiles for each of them.

Reply:
I'm happy to publish just one. I did, though, include a Dockerfile example for each OS. That could go in the doc I suppose.

Reply:
I think people who use Kali will need to know that they need to Install Python (they'll see this as soon as `pip` fails). I think we have to careful about exploding out distro based docs/requirements if we can avoid it.

Reply:
Yes let's publish just one.